### PR TITLE
JS: make string formatting a taint step 

### PR DESCRIPTION
**Pull request by esben-semmle**
This PR adds string formatting, like `util.format('echo "%s"', x)`, as a default taint step.

There are no new results on the standard benchmarks, but performance is at least [unchanged](https://git.semmle.com/gist/esben/b975f429cd3f703095b3f2ab9a7f011d).

### CODE REVIEW

File: javascript/ql/src/semmle/javascript/frameworks/StringFormatters.qll [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+ * A printf-style call that substitutes the embedded format specifiers of a format string for the format arguments.
+ */
+abstract class PrintfStyleCall extends DataFlow::CallNode {
+  /**
+   * Gets the format string.
+   */
+  abstract DataFlow::Node getFormatString();
+
+  /**
+   * Gets the i'th argument to the format string.
```
Comment (xiemaisi) [inline javascript/ql/src/semmle/javascript/frameworks/StringFormatters.qll]: `i`th (cf https://wiki.semmle.com/display/SSG/nth)

**Comment by esben-semmle** [comment]
Amended.

**Comment by xiemaisi** [comment]
Needs conflict resolution (the web interface actually works reasonably well).

**Comment by esben-semmle** [comment]
Resolved.

**Pull request merged**